src/com/lftechnology/java/training/niraj/ems/services/CrudService.java

```diff
@@ -0,0 +1,38 @@
+package com.lftechnology.java.training.niraj.ems.services;
+
+import java.sql.SQLException;
+
+import com.lftechnology.java.training.niraj.ems.exceptions.CustomException;
+
+public interface CrudService<T, S> {
+
+    /**
+     * Saves information
+     * 
+     * @author Niraj Rajbhandari <nirajrajbhandari@lftechnology.com>
+     * @param t
+     * @return
+     * @throws Exception
+     */
+    public T save(T t) throws SQLException;
+
+    /**
+     * Finds a record
+     * 
+     * @author Niraj Rajbhandari <nirajrajbhandari@lftechnology.com>
+     * @param s
+     * @return
+     * @throws Exception
+     */
+    public T find(S s) throws SQLException;
+
+    /**
+     * Updates a record
+     * 
+     * @author Niraj Rajbhandari <nirajrajbhandari@lftechnology.com>
+     * @param t
+     * @return
+     * @throws Exception
+     */
+    public boolean update(T t) throws SQLException,CustomException;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this method to throw at most one checked exception instead of: java.sql.SQLException, !unknownSymbol! [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1160)